test/usecases/recursion/recursion.mod

```diff
@@ -0,0 +1,34 @@
+NEURON {
+    SUFFIX recursion
+}
+
+: single recursive call
+FUNCTION f_simple(n) {
+    f_simple = f_simple(n)
+}
+
+: multiple recursive calls
+FUNCTION fc(n) {
+    fc = fc(fc(fc(fc(n))))
+}
+
+: multiple recursive calls with different args
+FUNCTION fd(n) {
+    fd = fd(fc(fd(n) + fc(fd(n))))
```

Comment:
Every call to `fd` calls `fd` 3 times; and `fc` (in itself infinitely recursing). Hence, same question what does this test?

Reply:
Different arguments mainly, it's again there to make sure that NMODL compiles the mod file successfully (whether or not it's a good idea to actually run this function is a different issue). I put several tests just to see if I can break the codegen somehow.